ortho_seq_code/orthogonal_polynomial.py

```diff
@@ -317,18 +355,179 @@ def orthogonal_polynomial(filename, molecule, phenotype, sites, dm, pop_size, po
+        naming = os.path.basename(f.name)
+        np.save(os.path.join(out_dir, naming + str('_cov')), cov)
+
+        Pa = np.array(
+            [[[0.0 for z in range(dm)]
+                for j in range(pop_size)] for i in range(sites)])
+        reg11 = np.array(
+            [[[[0.0 for z in range(dm)] for i in range(dm)]
+                for j in range(sites)] for k in range(sites)])
+        # Regression of site k on site l
+        # reg11[2][1] is a matrix (hence two more indices)
+        # containing the regressions of
+        # each element of the site 2 vector on each element of the site 1 vector
+        for k in range(sites):
+            for l in range(sites):
+                for i in range(0, dm):
+                    for j in range(0, dm):
+                        if var[l][j] > 0.0000000000001:
+                            reg11[k][l][i][j] = cov[k][l][i][j] / var[l][j]
+                        else:
+                            reg11[k][l][i][j] = 0
+        print("computed reg11")
+        naming = os.path.basename(f.name)
+        np.save(os.path.join(out_dir, naming + str('_reg11')), reg11)
+
+        # # # # # First order terms with zeros except for the value that is
+        # # # # # present (i.e. orthogonalized within each vector)
+        for k in range(sites):  # site
+            for i in range(pop_size):  # indiv
+                Pa[k][i] = sr.inner_general(
+                    sr.outer_general(phi[k][i], phi[k][i]), P[k][i])
+        print("computed Pa: first order orthogonalized within each vector")
+        naming = os.path.basename(f.name)
+        np.save(os.path.join(out_dir, naming + str('_Pa')), Pa)
+
+        P1i1 = np.array(
+            [[[[0.0 for z in range(dm)] for i in range(pop_size)]
+                for j in range(sites)]
+                for k in range(sites)])
+        for i in range(0, pop_size):  # Individuals
+            for j in range(sites):
+                for k in range(sites):
+                    if k != j:
+                        P1i1[j][k][i] = P[j][i] - sr.inner_general(
+                            reg11[j][k], Pa[k][i])
+        P2i1 = P1i1[1][0]
+        print("computed P1i1")
+        naming = os.path.basename(f.name)
+        np.save(os.path.join(out_dir, naming + str('_P1i1')), P1i1)
+
+        # # # # Variance in P2i1
+        varP1i1 = np.array(
+            [[[0.0 for z in range(dm)]
+                for i in range(sites)] for j in range(sites)])
+        for i in range(0, pop_size):  # individuals
+            for j in range(sites):
+                for k in range(sites):
+                    if k != j:
+                        varP1i1[j][k] += (P1i1[j][k][i] ** 2) / pop_size
+        print("computed varP1i1")
+        naming = os.path.basename(f.name)
+        np.save(os.path.join(out_dir, naming + str('_varP1i1')), varP1i1)
+        # # # # cov11i1[j][k][l] = cov between site j and (site k independent of l)
+        cov11i1 = np.array(
```

Comment:
a better variable name would be  cov_11_i1 same for var

Reply:
I think maybe it is better to keep the letters and numbers together since when do the naming, I'm using the '_' to name the object accordingly as it gets saved. But I agree, it does look weird!